receiver/loadgenreceiver/metrics_test.go

```diff
@@ -60,3 +66,46 @@ func TestMetricsGenerator_doneCh(t *testing.T) {
 		})
 	}
 }
+
+func TestMetricsGenerator_TSDBWorkaround(t *testing.T) {
```

Comment:
[to reviewer] I know this doesn't look the best, but it seems to be the most straightforward way to test given the current generator implementation